### PR TITLE
Users frontend

### PR DESCRIPTION
**Pull request by nikhitashanker**
- Basic Profile Page ~ create/edit profile
Screen Shot 2020-10-10 at 3 30 03 PM
Screen Shot 2020-10-10 at 3 30 14 PM

- Register for EWI Page ~ create/edit profile
Same as basic profile page in addition to:
Screen Shot 2020-10-10 at 3 33 31 PM
Screen Shot 2020-10-10 at 3 33 39 PM

- Events + Users Summary Page ~ admin viewable summary of events user is attending
Screen Shot 2020-10-10 at 3 32 36 PM

Known issue to be addressed in future PR: company choices not updating

### CODE REVIEW

**Comment by nikhitashanker** [comment]
Register for EWI Page; Basic Profile Page; Events + Users Summary Page; User session creation

User session parts that are not final have been commented out
For reference pages completed this quarter are on the following routes:
YES /register : Page for signing in with Google account
YES /registerbasic: Page for filling out the users profile for each users
YES /registerEWI : Page for filling out the EWI profile for each users
YES ADMIN /users/admin: Page for reporting the number of events each student has attended

**Comment by nikhitashanker** [comment]
I will create a new pull request once user session is complete

File: client/src/components/EditProfile.js [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      initial_first_name: null,
+      first_name: null,
+
+      initial_last_name: null,
+      last_name: null,
+
+      initial_phone_number: null,
+      phone_number: null,
+
+      //can't change the email
```
Comment (nikhitashanker) [inline client/src/components/EditProfile.js:44]: Remove comment.

File: client/src/components/Register.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+				.then(result => {
+					console.log(result);
+					let user_data = { ...result[0].data[0] };
+					let user_occupations = [...result[1].data];
+					let user_majors = [...result[2].data];
+
+					// console.log(user_data);
+					// console.log(user_occupations);
+					// console.log(user_majors);
+
```
Comment (nikhitashanker) [inline client/src/components/Register.js:69]: can remove this.

File: server/routes/users.js [hunk 3]
```diff
@@ -65,12 +65,15 @@ router.get('/ids', function(req, res, next) {
 
 // Add a user
 router.post('/register', function(req, res, next) {
+  //TODO: fix inconsistency b/w id and ID*S*
```
Comment (nikhitashanker) [inline server/routes/users.js:68]: Remove TODO comment

File: server/routes/users.js [hunk 4]
```diff
@@ -178,6 +147,22 @@ router.post('/register', function(req, res, next) {
...
+  }
+
+  // Production environment session setup
+  if (app.get('env') === 'production') {
+    app.set('trust proxy', 1) // trust first proxy
+    sess.cookie.secure = true // serve secure cookies
+  }
+
+  app.use(session(sess))
+
```
Comment (nikhitashanker) [inline server/routes/users.js:165]: Remove.

File: client/src/components/EditProfile.js [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+      })
+      .catch(err => console.log(err));
+    /*
+    axios.get('/majors/ids')
+      .then(result => {
+        let mids = result.data.map(function(major) { return major.id });
+        this.setState({ mids: mids });
+      })
+      .catch(err => console.log(err));
+      */
```
Comment (nikhitashanker) [inline client/src/components/EditProfile.js:98]: Remove.

File: server/routes/users.js (hunk 3 above)
Comment (nikhitashanker) [reply server/routes/users.js:68]: Ignore inconsistency still there.

**Review by samritha-n — APPROVED** [review]

**Review by christinatong01 — APPROVED** [review]

**Pull request merged**